### ISSUE TITLE
Babel 6 support

### ISSUE THREAD

**Issue by thealjey** [RReverser/babel-plugin-uglify#3]
http://babeljs.io/blog/2015/10/29/6.0.0

**Comment by thealjey** [comment]
I just created a PR #4 for this

**Comment by RReverser** [comment]
For those who wonder about the status, see https://github.com/RReverser/babel-plugin-uglify/pull/4#issuecomment-170895081. PRs welcome.

**Comment by fregante** [comment]
Until then, on the command line you can run `babel script.js | uglify > script.min.js`

But I'd love a cssnano-like babel-based minifier at some point, hopefully somebody is already working on it, some smaller plugins already exist: https://babeljs.io/docs/plugins/#minification

**Edit:** they did it! [babili](https://github.com/babel/babili)

**Comment by rochapablo** [comment]
I've tried ```babel script.js | uglify > script.min.js``` but it's not working.

```
$ babel script.js | uglify > script.min.js
bash: uglify: command not found
You have mistakenly installed the `babel` package, which is a no-op in Babel 6.
Babel's CLI commands have been moved from the `babel` package to the `babel-cli` package.
```

```
$ npm run babel-cli script.js | uglify > script.min.js
bash: uglify: command not found
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "babel-cli" "script.js"
npm ERR! node v5.10.0
npm ERR! npm  v3.8.3

npm ERR! missing script: babel-cli
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     D:\www\noonii\usuario\npm-debug.log
```

**Comment by RReverser** [comment]
@rochapablo Error is descriptive enough "bash: uglify: command not found". You forgot to install it :)

**Comment by rorydownes** [comment]
Is this plugin really necessary?

Minification seems to work by default by simply adding minified: true option on task options.

e.g)

```
options: {
        minified: true,
        sourceMap: true,
        plugins: ['transform-react-jsx'],
        presets: ['es2015', 'react']

      },
```

**Comment by fregante** [comment]
It's not nearly the same; minified only takes care of whitespace if I remember correctly, it's not a real minifier (like babili would be)

**Comment by shnigi** [comment]
I just can't get this to work:
Warning: Unknown plugin "uglify:after" specified in "/Users/path/poc_app/.babelrc" at 0, attempted to resolve relative to "/Users/path/poc_app" Use --force to continue.

**Comment by qm3ster** [comment]
Discontinued in favor of babili?

**Comment by RReverser** [comment]
@qm3ster Nope, just not supported for Babel 6 due to breaking changes described earlier. Although, after these years, there are moves to provide layer for ESTree in Babel 7 or even completely go to ESTree, so might be possible that it will be working again.

**Comment by ftorrado** [comment]
@bfred-it 
> Until then, on the command line you can run babel script.js | uglify > script.min.js

Is there any way this would work if I want to generate a single source map from running both babel and uglify?

**Comment by RReverser** [comment]
@ftorrado Sure, you can use `--in-source-map` option of UglifyJS for that with either `inline` option for piping or explicit filename if you're running it as separate commands. Although that question is not really in scope of this project :)
